// @shopify/polaris/build/esm/components/Modal/components/Footer/Footer.js
import React from 'react';
import { buttonsFrom } from '../../../Button/utils.js';
import { Inline } from '../../../Inline/Inline.js';
import { Box } from '../../../Box/Box.js';

function Footer({
  primaryAction,
  secondaryActions,
  children
}) {
  const primaryActionButton = primaryAction && buttonsFrom(primaryAction, {
    primary: true
  }) || null;
  const secondaryActionButtons = secondaryActions && buttonsFrom(secondaryActions) || null;
  const actions = primaryActionButton || secondaryActionButtons ? /*#__PURE__*/React.createElement(Inline, {
    gap: "2"
  }, secondaryActionButtons, primaryActionButton) : null;
  return /*#__PURE__*/React.createElement(Inline, {
    blockAlign: "center"
  }, /*#__PURE__*/React.createElement(Box, {
    borderBlockStart: "divider",
    minHeight: "var(--p-space-16)",
    padding: "4",
    width: "100%"
  }, /*#__PURE__*/React.createElement(Inline, {
    blockAlign: "center",
    align: "space-between"
  }, /*#__PURE__*/React.createElement(Box, null, children), actions)));
}

export { Footer };
